fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LucenePartitioner.java

```diff
@@ -261,36 +249,32 @@ public <M extends Message> CompletableFuture<Integer> addToAndSavePartitionMetad
         if (!isPartitioningEnabled()) {
             return CompletableFuture.completedFuture(null);
         }
-        return addToAndSavePartitionMetadata(
-                groupingKey,
-                getPartitioningTimestampValue(newRecord));
+        return addToAndSavePartitionMetadata(groupingKey, toPartitionKey(newRecord));
     }
 
     /**
-     * add a timestamp to the metadata of a given partition and save to db.
-     * The <code>count</code> will be incremented, and the <code>from</code> or <code>to</code> timestamps will
+     * add a partition field value to the metadata of a given partition and save to db.
+     * The <code>count</code> will be incremented, and the <code>from</code> or <code>to</code> boundary values will
      * be adjusted if applicable.
      *
-     * @param groupKey  grouping key
-     * @param timestamp document timestamp
+     * @param partitioningKey partitioning key
      * @return assigned partition id
      */
     @Nonnull
-    private CompletableFuture<Integer> addToAndSavePartitionMetadata(@Nonnull final Tuple groupKey, @Nonnull final Long timestamp) {
-        return getOrCreatePartitionInfo(groupKey, timestamp).thenApply(assignedPartition -> {
+    private CompletableFuture<Integer> addToAndSavePartitionMetadata(@Nonnull final Tuple groupingKey, @Nonnull final Tuple partitioningKey) {
+        return getOrCreatePartitionInfo(groupingKey, partitioningKey).thenApply(assignedPartition -> {
```

Comment:
Just to confirm, everywhere you use `partitioningKey` it is both the value, and the primary key, correct?

Reply:
Correct